shared/pubsub.js

```diff
@@ -205,8 +205,18 @@ export function createClient (url: string, options?: Object = {}): PubSubClient
   return client
 }
 
-export function createMessage (type: string, data: JSONType): string {
-  return JSON.stringify({ type, data })
+export function createMessage (type: string, data: JSONType, meta: ?Object): Object {
+  const message = { ...meta, type, data }
+  const stringify = function () {
+    return JSON.stringify(this)
+  }
```

Comment:
Why delay the stringification in a function? Why not stringify here and return the string instead?

Reply:
One reason is that, the way things are now, a function call is expected (called `[Symbol.toPrimitive]`). This change was made, as we discussed on Monday, so that we're able to shorten messages that are too long for push notifications. As a reminder, we wanted to be able to use regular objects that could be converted to a string on demand while still preserving the input, to be able to elide fields and shorten the message.

Now, as for why this wasn't made like:

```js
const string = JSON.stringify(message)
const stringify = function () {
    return message
}
```

I guess there's no strong reason, but it seemed appropriate to delay stringification until it's actually needed, if it's needed at all. In some cases, we call `JSON.stringify()` directly on the message (see `backend/pubsub.js`), and this means we'd be unnecessarily stringifying it twice.

As for why it's not the following (or the equivalent using a closure):

```js
let string
const stringify = function () {
    if (!string) string = JSON.stringify(this)
    return message
}
```

The reason is that it felt simpler the way it is now, but I could cache the value like that. I believe in most cases other than when broadcasting, stringification will be called once. For broadcasting, I fixed a bug in the previous code that called stringify every time (but as stated earlier, broadcast uses `JSON.stringify()` directly).

Reply:
Used the cached version